srtcore/api.cpp

```diff
@@ -683,6 +683,12 @@ int CUDTUnited::newConnection(const SRTSOCKET listen, const sockaddr_any& peer,
            // For redundancy group, at least, update the status in the group
            CUDTGroup* g = ns->m_IncludedGroup;
            ScopedLock glock (g->m_GroupLock);
+           if (g->m_bClosing)
+           {
+               error = 1; // "INTERNAL REJECTION"
+               goto ERR_ROLLBACK;
```

Comment:
I know this goto label already exists, but somehow all these gotos are better to be rewritten to something more readable.

Reply:
Will be. Not of course that goto in C++ still honors all required destructors, so it's not that dangerous if you use scoped tools.

Reply:
> For a number of years I have been familiar with the observation that the quality of programmers is a decreasing function of the density of go to statements in the programs they produce. More recently I discovered why the use of the go to statement has such disastrous effects, and I became convinced that the go to statement should be abolished from all "higher level" programming languages (i.e. everything except, perhaps, plain machine Code). 

Edsger W. Dijkstra. [Letters to the editor: go to statement considered harmful](https://dl.acm.org/doi/10.1145/362929.362947)

Reply:
I know. Just note that I wasn't actually the first to introduce this label and gotos in this function. OTOH note that goto in C++ is also under heavy restrictions and there are only extremely distinct cases when the use of it might make sense. And I believe this is the case here.